scripts/top-search-queries/get-top-search-results.go

```diff
@@ -0,0 +1,242 @@
+	InputFileName string `json:"inputFileName"`
+	NumResults    int    `json:"numResults"`
+}
+
+// main reads in a list of queries from a text file. These are intended to be the queries that are most commonly used
+// in the live Search API service.
+//
+// It then creates an output CSV file and processes the queries, by calling the live Search API, and outputs
+// all the query results to that CSV file.
+//
+// Each query is called with NLP (Natural Language Processing) initially switched off, then called again with
+// NLP switched on. The top 10 results of each query call are added to the output CSV file.
+func main() {
+	ctx := context.Background()
+	log.Info(ctx, "starting script to get results of most common queries from the Search API")
+
+	config := runConfig{}
+	flag.StringVar(&config.APIURL, "api_url", "https://api.beta.ons.gov.uk/v1", "the base url for the search api")
+	flag.StringVar(&config.InputFileName, "input_file_name", "search-queries.txt", "name of input file including extension")
+	flag.IntVar(&config.NumResults, "num_results", 10, "number of results to fetch for each query")
+	flag.Parse()
+	logData := log.Data{"config: ": config}
+	log.Info(ctx, "parsed config", logData)
+	run(ctx, config)
+	log.Info(ctx, "end of script")
+}
+
+func run(ctx context.Context, config runConfig) {
+	listOfQueries, err := readQueriesFromFile(ctx, config)
+	check(ctx, "failed reading queries file", err)
+
+	csvFile := createOutputCSVFile(ctx)
+	defer func() {
+		if err = csvFile.Close(); err != nil {
+			log.Fatal(ctx, "failed closing csv file", err)
+		}
+	}()
+
+	log.Info(ctx, "make each request to the Search API with NLP off initially, then NLP on")
+	for _, query := range listOfQueries {
+		searchResp := getQueryResults(ctx, query, "false", config)
+		AddResultsToCSV(ctx, query, csvFile, "false", searchResp)
+		searchResp = getQueryResults(ctx, query, "true", config)
+		AddResultsToCSV(ctx, query, csvFile, "true", searchResp)
+	}
+}
+
+// createOutputCSVFile creates the CSV file ready for the query results to be added to. It firstly adds a header row.
+func createOutputCSVFile(ctx context.Context) *os.File {
+	logData := log.Data{"CSV name: ": "top-search-query-results.csv"}
+	csvFile, err := os.Create("top-search-query-results.csv")
+	check(ctx, "failed creating file", err)
+	log.Info(ctx, "successfully created csv file", logData)
+	writeHeaderRow(csvFile, ctx)
+	return csvFile
+}
+
+// getQueryResults calls the Search API for a particular query string and NLP weighting (true or false)
+// If the NLP weighting is true then NLP is used, if false then NLP is not used. The results of the query call are each
+// written to the supplied output CSV file as separate rows.
+func getQueryResults(ctx context.Context, querySupplied, nlpWeighting string, cfg runConfig) searchResponse {
+	sdkItems := callSearchAPI(ctx, querySupplied, nlpWeighting, cfg.APIURL, cfg.NumResults)
+
+	searchItems := make([]searchItem, 0)
+	for i := range sdkItems {
```

Comment:
Go allows you to range over values, not just indexes which can make the code more readable
```suggestion
	for i, sdkItem := range sdkItems {
```
or 
```suggestion
	for _ , sdkItem := range sdkItems {
```

Reply:
NB. This comment's not critical, the code is functional as is I just think it is a little cleaner.

Reply:
Linter recommends otherwise. I'll retract this comment.